### DISCUSSION TITLE
[Question] Can we hide Dimensions from dimensions list?

### DISCUSSION THREAD

**Discussion by EMREOYUN** [popstarfreas/Dimensions#75] [Q&A]

**Comment by EMREOYUN** [comment]
Nevermind, i actually managed to do this but i don't how to write in TypeScript language to add merge, so any helps?

**Reply by popstarfreas** [reply]
Have you tried setting up a development copy via https://github.com/popstarfreas/Dimensions#development ?
Then when you edit the typescript and run `npm run build` it will build it to javascript. Then you can `npm run start` to run dimensions and you check that what you changed is working.

**Reply by EMREOYUN** [reply]
This is the reason, i don't know anything about TypeScript, i'm working on JavaScript when i need to change something on Dimensions.

**Reply by popstarfreas** [reply]
Typescript is the same as Javascript, it just has more features. Anything you write in Javascript is valid Typescript. But I would say you should try to, when writing Typescript as though you are writing Javascript, annotate types in methods and functions.

**Reply by EMREOYUN** [reply]
Okay, i'll try making a port to TypeScript and if it is works, i'll create a pull request. 
I'll quickly explain how this system works:
- Basically there is boolean "hidden" setting at all dimensions in config. 
- When it is set to true, the dimension will not shown up in /dimensions list(even hidden in the server list shows on kicked) but typing exact server name still passes you to hidden server.
- When it is set to false, the dimension will shown up in /dimensions as normal and players still can go there.

**Idea:** In my server, my hidden servers uses a special whitelist function and only really specified users(not IP addresses as in original TShock) can enter this hidden server, other users will kicked and see the kick message in chat and Dimensions asks them to go to visible servers in the list
